2-mandatory/3-financial-times.js

```diff
@@ -6,32 +6,125 @@
 */
 function potentialHeadlines(allArticleTitles) {
     // TODO
+
+//1.filter()method
+return  allArticleTitles.filter(letters => letters.length<=65);
+    
+//2.for ... of loop
+//      let headlines = [];
+//      for(let title of allArticleTitles){
+//         if (title.length <=65){
+//             headlines.push(title);
+//         }
+//         return headlines
+//      }
+
 }
 
 /*
     The editor of the FT likes short headlines with only a few words!
-    Implement the function below, which returns the title with the fewest words.
-    (you can assume words will always be seperated by a space)
+    Implement the function below, which returns the title with the fewest
+    words.
+    (you can assume words will always be separated by a space)
 */
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
+    // TODO: 
+     //Barath's solution:
+    // declare a function in advance to return the number of every element in the array:
+    
+
+    //  function numberOfWords(title){
+    //     return title.split(' ').length;
+    //  }
+
+    // function titleWithFewestWords(allArticleTitles) {
+    //   let lowestNumberOfWords;
+    //   let titleWithFewestWords;
+
+    //   for (let title of allArticleTitles) {
+    //     // working out the number of words in the title by splitting on the space 
+    //     // character
+    //     // this will generate an array. Read more: 
+    //     //https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split
+    //     let numWords = numberOfWords.title(" ").length;
+        
+    //     if (
+    //       lowestNumberOfWords === undefined ||
+    //       numWords < lowestNumberOfWords
+    //     ) {
+    //       lowestNumberOfWords = numWords;
+    //       titleWithFewestWords = title;
+    //     }
+    //   }
+
+    //   return titleWithFewestWords;
+    // }
+
+
+   let wordNums =[];
+   for (let i=0; i<allArticleTitles.length; i++){
+       //return a new array of every article's length:
+       wordNums.push(allArticleTitles[i].split(' ').length)
+   }
+   //return the array with lowest number using indexOf()method:
+   return allArticleTitles[wordNums.indexOf(Math.min(...wordNums))]
 }
 
 /*
-    The editor of the FT has realised that headlines which have numbers in them get more clicks!
+    The editor of the FT has realized that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
 function headlinesWithNumbers(allArticleTitles) {
     // TODO
+    //1. using 'regex':
+    let  NumberArray = [];
+    const regex = /[0-9]/g;
```

Comment:
This is a good implementation. Think if you could do this simpler way just using for loop and if else.

Reply:
Yes, I had optional solution with for of loop +  if condition syntax btw line 91 and 110 & commented it